### DISCUSSION TITLE
Using AND_* for Loras?

### DISCUSSION THREAD

**Discussion by Disguised-Giraffe** [ljleb/sd-webui-neutral-prompt#45] [General]
Is there any known use of using the Neutral Prompt AND_*'s with Loras? If anyone has experience with these two I'd be interested to know if anything particularly works well.

For example would:  "..., <lora: a:1> AND_PERP <lora: b:1>, ..." have any effect or good use? If the AND_ had an effect on lora's just by themselves I think I'd get a lot of use out of the extension. I like to combine lora's and do a lot of inpainting. I run in to conflicting lora's or just have too many in one pass and often have to reapproach my prompts and workflow.
I tend to keep my lora's and tag words separated to the bottom of my prompt. So, with the AND function would it be better to avoid that, inserting lora's with their relevant text? ex: "interior of a haunted gothic dungeon room <lora:Isometric_Maps_Lora:1> AND_PERP a large wooden treasure chest on top a pile of gold coins <lora:Game_Icon_Lora:1>"
And there's maybe nesting as well? Interested if anyone has advice for using this ext. with lora's for the art or help prevent lora's contaminating an image.

**Comment by ljleb** [comment]
To use different loras in different prompts, you can use the composable lora extension:
https://github.com/a2569875/stable-diffusion-webui-composable-lora

The extensions should be intercompatible. Let me know if you run into any issues. Personally it did not give good results, but maybe you'll have more luck.

Composable lora normally works with AND prompts, but you can use any of the keyword of the extension. In the background, before sending the prompts to the webui and composable lora, the extension will convert any `AND_*` keyword to `AND` and will discard any brackets for nesting `[` and `]`. This gives you the order in which composable lora sees the sub-prompts.
